### DISCUSSION TITLE
Ask to replace duplicated files

### DISCUSSION THREAD

**Discussion by omnigenous** [gokcehan/lf#1795] [Q&A] [answered]
Is there a way to make `lf` ask to replace files with duplicated names when moving or coping to other directory instead of adding `.~N~` suffix?

**Comment by catthegirl** [comment]
I don't believe that there is a way without reconfiguring the paste command, however, you're more than welcome to write your own behavior for the paste command inside your lfrc file.

**Reply by omnigenous** [reply]
I read https://github.com/gokcehan/lf/blob/master/doc.md#file-operations but I don't have skills to write my own implementation.

I managed to modify `paste` example to loop over `$list` to process files individually. Haven't figured out how to test if file name is duplicated but I will with time.

I want to add dialog that would ask user what to do with duplicates. Ex: "Duplicated file name found `FILENAME`.  What to do? [k]eep old, [r]eplace with new or [p]reserve both"

For keep I would do nothing
For replace `cp`
For preserve both I like how `lf` doesn't but not sure how to implement that in shell (`cp -a` is not the same)

I know I can use `echo` to render text at bottom left corner of  `lf` but I don't know how to make `lf` read `k`, `r` and `p` when this text is rendered. Also is there a way to clean/reset bottom left corner after `echo`?

**Reply by omnigenous** [reply]
```sh
cmd paste %{{
    load=$(cat ~/.local/share/lf/files)
    mode=$(echo "$load" | sed -n '1p')
    list=$(echo "$load" | sed '1d')

    for path in $list; do
      file_name=$(basename $path)
      new_path="$PWD/$file_name"
      
      if [ -f "$new_path" ]; then
        echo "Duplicated file name found $file_name. What to do? [k]eep old, [r]eplace with new or [p]reserve both"
        
        # --- Wait for answer ---
        
        # --- Handle answer ---
        # if [ $mode = 'copy' ]; then
        #   cp -R $list .
        # elif [ $mode = 'move' ]; then
        #   mv $list .
        #   rm ~/.local/share/lf/files
        #   lf -remote 'send clear'
        # fi
        
        # --- Reset bottom left ---
      fi
    done
}}
```

**Comment by DusanLesan** [comment, marked as answer]
I took your command and changed it a bit. It does not work exactly like default because I have different `dupfilefmt` preferences, but you might be able to tweak that.
```bash
cmd paste %{{
	load=$(cat ~/.local/share/lf/files)
	mode=$(echo "$load" | sed -n '1p')
	list=$(echo "$load" | sed '1d')

	for path in $list; do
		file_name=$(basename $path)
		new_path="$PWD/$file_name"

		if [ -f "$new_path" ]; then
			echo "Duplicated file name found $file_name. What to do? [k]eep old, [r]eplace with new or [p]reserve both "
			read ans

			if [ $ans = "k" ]; then
				continue
			elif [ $ans = "r" ]; then
				echo "replacing $new_path with $path"
			elif [ $ans = "p" ]; then
				[[ "$file_name" == .* ]] && { dot="."; file_name="${file_name#.}"; }
				name="${file_name%%.*}"
				ext="${file_name#"$name"}"
				i="${name##*[!0-9]}"
				[ -z "$i" ] && i=1 || name="${name%%$i*}"
				while [ -e "$PWD/$dot$name.$i$ext" ]; do
					((i++))
				done
				new_path="$PWD/$dot$name.$i$ext"
			else
				notify-send "Unrecognized input"
			fi
		fi

		if [ $mode = 'copy' ]; then
			cp -R "$path" "$new_path"
		elif [ $mode = 'move' ]; then
			mv "$path" "$new_path"
		fi
	done

	lf -remote "send $id clear"
}}
```
